compiler/rustc_passes/src/dead.rs

```diff
@@ -447,6 +447,8 @@ impl<'v, 'k, 'tcx> ItemLikeVisitor<'v> for LifeSeeder<'k, 'tcx> {
     fn visit_impl_item(&mut self, _item: &hir::ImplItem<'_>) {
         // ignore: we are handling this in `visit_item` above
     }
+
+    fn visit_foreign_item(&mut self, _item: &'v hir::ForeignItem<'v>) {}
```

Comment:
This is currently incorrect, but already was so before this PR.

```rust
#[repr(C)]
struct Type(u8);

extern "C" {
    #[allow(dead_code)]
    fn hey(t: Type);
}
```
shouldn't warn for `Type` because type is used by `fn hey`.
Should probably open a separate issue for this unless you quickly want to fix this in a followup PR

Reply:
Investigating in a follow-up PR.